Mathlib/CategoryTheory/Galois/GaloisObjects.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.Galois.Basic
+import Mathlib.CategoryTheory.Limits.FintypeCat
+import Mathlib.CategoryTheory.Limits.Preserves.Limits
+import Mathlib.CategoryTheory.Limits.Shapes.SingleObj
+import Mathlib.Logic.Equiv.TransferInstance
+
+/-!
+# Galois objects in Galois categories
+
+We define Galois objects in a Galois category `C` in a fibre functor independent way and show
+equivalent characterisations.
+
+## Main definitions
+
+* `GaloisObject` : Connected object `X` of `C` such that `X / Aut X` is terminal.
+
+## Main results
+
+* `galois_iff_pretransitive` : A connected object `X` is Galois if and only if `Aut X`
+                               acts transitively on `F.obj X` for a fibre functor `F`.
+
+-/
+universe u₁ u₂ v₁ v₂ v w
+
+namespace CategoryTheory
+
+namespace PreGaloisCategory
+
+open Limits Functor
+
+noncomputable instance {G : Type v} [Group G] [Finite G] :
+    PreservesColimitsOfShape (SingleObj G) FintypeCat.incl.{w} := by
+  choose G' hg hf e using Finite.exists_type_zero_nonempty_mulEquiv G
+  exact Limits.preservesColimitsOfShapeOfEquiv (Classical.choice e).toSingleObjEquiv.symm _
+
+/-- A connected object `X` of `C` is Galois if the quotient `X / Aut X` is terminal. -/
+class GaloisObject {C : Type u₁} [Category.{u₂, u₁} C] [GaloisCategory C] (X : C)
```

Comment:
In names below, you already abbrev this to `galois`. So maybe this name could also be `Galois` or `IsGalois`?

Reply:
I renamed it to `IsGalois` to align it with `IsGalois` from `Mathlib.FieldTheory.Galois`, I think we should then also rename `ConnectedObject` to `IsConnected` in a later PR.